src/fides/api/ops/service/messaging/message_dispatch_service.py

```diff
@@ -411,6 +415,49 @@ def _mailgun_dispatcher(
         raise MessageDispatchException(f"Email failed to send due to: {Pii(e)}")
 
 
+def _twilio_email_dispatcher(
+    messaging_config: MessagingConfig,
+    message: EmailForActionType,
+    to: Optional[str],
+) -> None:
+    """Dispatches email using twilio sendgrid"""
+    if not to:
+        logger.error("Message failed to send. No email identity supplied.")
+        raise MessageDispatchException("No email identity supplied.")
+    if not messaging_config.details or not messaging_config.secrets:
+        logger.error(
+            "Message failed to send. No twilio email config details or secrets supplied."
+        )
+        raise MessageDispatchException(
+            "No twilio email config details or secrets supplied."
+        )
+
+    try:
+        sg = sendgrid.SendGridAPIClient(
+            api_key=messaging_config.secrets[
+                MessagingServiceSecrets.TWILIO_API_KEY.value
+            ]
+        )
+        from_email = Email(
+            messaging_config.details[MessagingServiceDetails.TWILIO_EMAIL_FROM.value]
+        )
+        to_email = To(to.strip())
+        subject = message.subject
+        content = Content("text/html", message.body)
+        mail = Mail(from_email, to_email, subject, content)
+        response = sg.client.mail.send.post(request_body=mail.get())
+        if response.status_code >= 400:
+            logger.error(
+                "Email failed to send with status code: %s", response.status_code
+            )
+            raise MessageDispatchException(
+                f"Email failed to send with status code {response.status_code}"
+            )
```

Comment:
might we want to look for an error message beyond just the status code coming back in the response here? not a huge deal, and not sure if their API makes that available to us, but may just make any debugging a bit easier.

Reply:
I don't want to expose possible pii here, so this is a defensive way to prevent this. 

I can use our `Pii()` wrapper here, so that we can at least debug in dev mode